auth-server/src/services/auth.ts

```diff
@@ -51,7 +51,41 @@ export default class AuthService {
 		}
 	}
 
-	public async SignUp(email, password, username, name): Promise<AuthObjectType> {
+	public async Logout(token: string, refreshToken: string) {
+		if (!refreshToken) {
+			throw new Error('refresh token not provided')
+		}
+
+		const refreshTokenObj = await RefreshToken
+			.query()
+			.where('token', refreshToken)
+			.first()
+
+		if (!refreshTokenObj) {
+			throw new ForbiddenError('Refresh token not found')
+		}
+
+		// verify a token asymmetric - synchronous
+		const decoded = jwt.verify(token, publicKey)
```

Comment:
don't you want to first check the token (a.k.a access token). Then query the refresh token from this `user_id` and invalidate anything you find (there should be only 1) ?

Reply:
Just realized that you actually can have several refresh token for the same user if you login with another browser/device or just delete cookies and local storage and login again. The side effect is that deleting all token at logout will logout this user from every device, we can live with that. Not super user friendly but this doesn't happen often that users logout I imagine.

Reply:
noted